Sources/PenguinStructures/Tuple.swift

```diff
@@ -63,6 +63,12 @@ public struct Tuple<Head, Tail: TupleProtocol>: TupleProtocol {
   
   /// The number of elements
   public static var count: Int { Tail.count + 1 }
+
+  /// Creates a tuple whose first element is `head` and whose remaining elements are `tail`.
```

Comment:
This sort of locks us into a cons-list representation.  If you want nice error messages as you did for the arrays, we should consider whether it would be better to use a flat representation.

Reply:
I'm encountering a lot of places where it's very convenient to recursively build tuples so that you can write a `Tuple -> Tuple` function that works on arbitrary tuples:
* the `VariableTuple` methods in this PR: https://github.com/borglab/SwiftFusion/pull/74/files#diff-5004d4f5a2ce54dee4f4facfeba18a17R35
* making `Tuple` conform to `AdditiveArithmetic` and `Differentiable` (this will be important for the next SwiftFusion PR I'm working on)

There isn't a way to write these things with a flat representation, other than generating separate code for each tuple, right?

I wouldn't be opposed to switching to the flat representation eventually. But I'd like to be able to use this representation now because it makes it so much easier to try things out.

Reply:
I think I agree.